new feature: givenItem-specific random ranges for givenAmount

adamdrakon opened the pull request:
Example usage in config.lua:

Config.Animals = {
    [-1143398950]  = { name = "Big Grey Wolf", givenItem = { "meat", "wolf_fang", "wolf_heart" }, givenAmount = { {2,5}, {1,2}, 1 }, money = 3, gold = 0, rolPoints = 0, xp = 3, poorQualityMultiplier = 1.0, goodQualityMultiplier = 1.5, perfectQualityMultiplier = 2, poor = 85441452, good = 1145777975, perfect = 653400939 },
}

The above example would make it so Big Grey Wolf carcasses give a random amount of "meat" between 2 and 5 and a random amount of "wolf_fang" between 1 and 2, and a guaranteed 1 "wolf_heart" when sold at a butcher.

adamdrakon committed:
new feature: givenItem-specific random ranges for givenAmount

Example usage in config.lua:

Config.Animals = {
    [-1143398950]  = { name = "Big Grey Wolf", givenItem = { "meat" }, givenAmount = { {2,5} }, money = 3, gold = 0, rolPoints = 0, xp = 3, poorQualityMultiplier = 1.0, goodQualityMultiplier = 1.5, perfectQualityMultiplier = 2, poor = 85441452, good = 1145777975, perfect = 653400939 },
}

The above example would make it so Big Grey Wolf carcasses give a random amount of "meat" between 2 and 5, when sold at a butcher.

outsider31000 commented:
Can you add an example on about 2 items.

outsider31000 commented:
I would assume givenAmmount ={ {2,5}, {1,3}} for second item ?

adamdrakon commented:
> 

Correct. I'll add to the example.

outsider31000 commented:
So in the config need some changes. The current method to be removed and an explanation there.

outsider31000 commented:
For the itemQuantity

adamdrakon commented:
I wouldn't remove the current method though, personally.
There's room for both to exist.

Let's say for example you have Config.ItemQuanity.Min set to 1 and Config.ItemQuanity.Max set to 3, and are happy to use 0 (to trigger the current random) in most cases, but for the case of just a few animals you'd like different random ranges in place for certain item rewards you can then use this new feature.

So the config file just needs some extra notation on how to use the new feature, in my opinion.

adamdrakon committed:
Notation update for using new random givenAmount feature.

Added notation on how to specify a random givenAmount  for a givenItem, independent of the range specified by Config.ItemQuantity.Min and Config.ItemQuantity.Max.

adamdrakon committed:
Notation update for using new random givenAmount feature. (typo fix)

Added notation on how to specify a random givenAmount  for a givenItem, independent of the range specified by Config.ItemQuantity.Min and Config.ItemQuantity.Max.

adamdrakon committed:
@adamdrakon Notation update for using new random givenAmount feature. (typo fix 2)

Added notation on how to specify a random givenAmount  for a givenItem, independent of the range specified by Config.ItemQuantity.Min and Config.ItemQuantity.Max.

adamdrakon commented:
Whoops, caught my avatar when copy pasting the commit short summary... my bad.

outsider31000 commented:
Oh i see.  Its just a bit confusing having both.
